benchmark/test/test_method.py

```diff
@@ -0,0 +1,53 @@
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+"""
+Method tests for the control service benchmarks.
+"""
+
+from zope.interface import Interface
+
+from flocker.testtools import TestCase
+
+from benchmark._method import validate_no_arg_method, InvalidMethod
+
+
+class ITest(Interface):
+    """
+    An interface for testing.
+    """
+
+    def noargs():
+        """
+        Method that takes no arguments.
+        """
+
+    def hasargs(a, b, c):
+        """
+        Method that takes arguments.
+        """
+
+
+class MethodTests(TestCase):
+
+    def noargs_is_noargs_method(self):
+        """
+        A no-argument method validates as no-arg method.
+        """
+        validate_no_arg_method(ITest, 'noargs')
+
+    def hasargs_fails_noargs_method(self):
+        """
+        An argument-taking method fails to validate as no-arg method.
+        """
+        exception = self.assertRaises(
+            InvalidMethod, validate_no_arg_method, ITest, 'hasargs'
+        )
+        self.assertIn('requires parameters', str(exception))
+
+    def not_present_fails_noargs_method(self):
+        """
+        A non-present method fails to validate as no-arg method.
+        """
+        exception = self.assertRaises(
+            InvalidMethod, validate_no_arg_method, ITest, 'notpresent'
+        )
+        self.assertIn('not found in interface', str(exception))
```

Comment:
You can probably configure pylint to do that actually. A thing I'd like to do (in my copious free time) is add a pylint check to our suite and gradually customize it to implement our style.